inference-engine/tests/functional/plugin/shared/include/behavior/infer_request.hpp

```diff
@@ -623,4 +624,39 @@ TEST_P(InferRequestTests, returnDeviceBusyOnGetPerformanceCountAfterAsyncInfer)
         std::cout << "Exception" << e.what() << std::endl;
     }
 }
-}  // namespace BehaviorTestsDefinitions
\ No newline at end of file
+
+class InferRequestTestsResultNotReady : public InferRequestTests {
+};
+
+TEST_P(InferRequestTestsResultNotReady, ReturnResultNotReadyFromWaitInAsyncModeForTooSmallTimeout) {
+    // Skip test according to plugin specific disabledTestPatterns() (if any)
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    // Create CNNNetwork from ngraph::Function
+    // return function which computes around 20ms on GNA SW
+    function = LayerTestsDefinitions::Basic_LSTM_S::GetNetwork(3000, 380);
+    InferenceEngine::CNNNetwork cnnNet(function);
+    // Load CNNNetwork to target plugins
+    auto execNet = ie->LoadNetwork(cnnNet, targetDevice, configuration);
+    // Create InferRequest
+    InferenceEngine::InferRequest req;
+    ASSERT_NO_THROW(req = execNet.CreateInferRequest());
+    InferenceEngine::ResponseDesc response;
+    InferenceEngine::StatusCode sts = InferenceEngine::StatusCode::OK;
+    auto testSuccesfull = false;
+    const auto maxAttempts = 1000;
+    const auto tooSmallTimeoutToComplete = 1; // 1ms
+    // we try to get RESULT_NOT_READY up to 1000 times
+    // as might happen that the result will be OK after waiting 1ms
+    // in some circumstances as thread scheduling, OS specifics etc
+    for (int tries = 0; tries < maxAttempts && !testSuccesfull; ++tries) {
+        req.StartAsync();
+        sts = req.Wait(tooSmallTimeoutToComplete);
+        if (sts == InferenceEngine::StatusCode::RESULT_NOT_READY) {
+            testSuccesfull = true;
```

Comment:
`testSuccesfull` -> `testSuccessfull`

Reply:
I suppose this test is sporadic. We cannot guarantee that one of 1000 times req is not ready. It can perform inference for all 1000 times for timeframe < 1 ms. In this case test will fail.

@myshevts @apankratovantonp do you agree?

Reply:
Yes, the inference can in theory complete _before_ call for the Wait(smalltimeout). This is invalid test

Reply:
Yes, it can fail sporadically, but the current model computes ~20ms (i5-8350U), so the risk of failure it very low even on much faster machines.
@myshevts @ilya-lavrenov So what is your recommendation to make a test for such scenario (RESULT_NOT_READY)?
Maybe we could produce a warning instead of a failure?
Or move this into some other target?
Or increase the computational time more to make it even lower probable to fail? (still could fail in theory)

Reply:
you check opposite way: add a callback to the request and capture the timestamp there. IF the timestamp val is larger than the timestamp taken AFTER return from the Wait(_no need for timeout_ do  the STATUS_ONLY) then you should  get NOT_READY. Of course before comparing, you should wait for callback to complete  first(some flag set in that).
this is the only valid cases, afaik.

Reply:
I relaxed this tests' assertions to make it deterministic. 
Please review.

Reply:
@ilya-lavrenov Could you approve or review?

Reply:
It does not look the same as @myshevts described...

Reply:
@ilya-lavrenov Please review the newest version. I guess it looks fine now (like @myshevts described).